src/waveform/renderers/waveformmarkset.h

```diff
@@ -19,13 +21,16 @@ class WaveformMarkSet {
     void clear();
 
     int size() const { return m_marks.size();}
-    WaveformMark& operator[] (int i) { return m_marks[i]; }
+    QSharedPointer<WaveformMark> operator[] (int i);
```

Comment:
That's the problem with an additional level of indirection and 
const-correctness ;)

The set itself is not modified, therefore const. But by handing out a 
mutable pointer the caller is able to modify individual elements afterwards. 
From a technical point of view 'const' would be correct. From a conceptual 
point of view it's difficult to decide. How strong is the association 
between the contents of each element and the set of pointers to those 
elements? The code says it's some kind of aggregation, but not really a 
composition.

On 21.08.2016 15:31, Ferran Pujol Camins wrote:
>
> In src/waveform/renderers/waveformmarkset.h 
> <https://github.com/mixxxdj/mixxx/pull/992#discussion_r75595250>:
>
> > @@ -19,13 +21,16 @@ class WaveformMarkSet {
> >      void clear();
> >
> >      int size() const { return m_marks.size();}
> > -    WaveformMark& operator[] (int i) { return m_marks[i]; }
> > +    QSharedPointer<WaveformMark> operator[] (int i);
>
> This methods are used to modify the WaveformMark. Should they be labeled 
> const if they modify one of the objects WaveformMarkSet stores?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/992/files/2f8863b7b8266565a97581fd38cfa6e87ab0ac36#r75595250>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-ai0O0i8dWbBD0dLvzGZlOIcimF4Rks5qiFM-gaJpZM4Jj7vm>.
>